README.md

```diff
@@ -4,4 +4,224 @@ GraphQL API Layer for Frappe Framework
+Data is returned based on Read access to the resource
+
+## Generic Mutations: set_value & save_doc
+- SetValue
+#### Query
+```graphql
+mutation SET_VALUE($doctype: String!, $name: String!, $fieldname: String!, $value: String!) {
+    setValue(doctype: $doctype, name: $name, fieldname: $fieldname, value: $value) {
+        doctype,
+        name,
+        fieldname,
+        value,
+        doc {
+            name,
+            ...on User {
+                first_name,
+                last_name,
+                full_name
+            }
+        }
+    }
+}
+```
+#### Variables
+```json
+{
+    "doctype": "User",
+    "name": "test@test.com",
+    "fieldname": "first_name",
+    "value": "Test X"
+}
+```
+#### Response
+```json
+{
+    "data": {
+        "setValue": {
+            "doctype": "User",
+            "name": "test@test.com",
+            "fieldname": "first_name",
+            "value": "Test",
+            "doc": {
+                "name": "test@test.com",
+                "first_name": "Test X",
+                "last_name": "Test A",
+                "full_name": "Test X Test A"
+            }
+        }
+    }
+}
+```
+
+## Pagination
+`limit_start` & `limit_page_length` could be passed to paginate through the doctypes
+#### Query
+```graphql
+{
+    User(limit_start: 0, limit_page_length: 10) {
+        name
+    }
+}
+```
+
+## Support Extensions via Hooks
+You can extend the SDLs with additional query / mutations / subscriptions. Examples are provided for a specific set of Scenarios. Please read [GraphQL Spec](http://spec.graphql.org/June2018/#sec-Object-Extensions) regarding Extending types. There are mainly two hooks introduced:
+- `graphql_sdl_dir`  
+Specify a list of directories containing `.graphql` files relative to the app's root directory.
+eg:
+```py
+# hooks.py
+graphql_sdl_dir = [
+    "./your-app/your-app/generated/sdl/dir1",
+    "./your-app/your-app/generated/sdl/dir2",
+]
+```
+The above will look for graphql files in `your-bench/apps/your-app/your-app/generated/sdl1` folder.
+- `graphql_schema_processors`  
+You can pass in a list of cmd that gets executed on schema creation. You are given `GraphQLSchema` object (please refer [graphql-core](https://github.com/graphql-python/graphql-core)) as the only parameter. You can modify it or extend it as per your requirements.
+This is a good place to attach the resolvers for the custom SDLs defined via `graphql_sdl_dir`
+
+### Examples
+### Adding a newly created DocType
+- Generate the SDLs in your app directory
+```bash
+$ bench --site test_site graphql generate_sdl --output-dir <your-app/graphql> --doctype <name>
+```
+- Specify this directory in `graphql_sdl_dir` hook and you are done.
+### Introducing a Custom Field to GraphQL
+- Add the `Custom Field` in frappe
+- Add the following to a `.graphql` file in your app and specify its directory via `graphql_sdl_dir`
+```graphql
+extend type User {
+    is_super: Int!
+}
+```
+
+### Adding a Hello World Query
```

Comment:
@faztp12 If you can please add an example of how the query can be requested for custom queries

Reply:
![image](https://user-images.githubusercontent.com/12441615/110754487-e5993000-826d-11eb-93f8-e7ec0fcb8ecf.png)
looks good ?

Reply:
@faztp12 Looks good, but I reckon adding a query with variables would make it comprehensive 💯 , what do you think?

Reply:
@assemmarwan updated and pushed, please check ;)

Reply:
Looks good 👍🏻